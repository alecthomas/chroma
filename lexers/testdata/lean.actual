def f(x : Nat) : Nat := x + 1

-- comment

theorem thm (p q r : Prop) : p ∧ (q ∨ r) → (p ∧ q) ∨ (p ∧ r) := by
  intro ⟨hp, hqr⟩
  show (p ∧ q) ∨ (p ∧ r)
  cases hqr with
  | inl hq =>
    have hpq : p ∧ q := And.intro hp hq
    apply Or.inl
    exact hpq
  | inr hr =>
    have hpr : p ∧ r := And.intro hp hr
    apply Or.inr
    exact hpr
