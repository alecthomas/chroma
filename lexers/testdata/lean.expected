[
  {"type":"KeywordNamespace","value":"def"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"f"},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"x"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Nat"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Nat"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"NameBuiltinPseudo","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"x"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"-- comment"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordNamespace","value":"theorem"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"thm"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"q"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"r"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordType","value":"Prop"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"q"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∨"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"r"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"→"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"q"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∨"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"r"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"NameBuiltinPseudo","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"by"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Name","value":"intro"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"⟨"},
  {"type":"Name","value":"hp"},
  {"type":"Operator","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hqr"},
  {"type":"NameBuiltinPseudo","value":"⟩"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Keyword","value":"show"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"q"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∨"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"("},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"r"},
  {"type":"Operator","value":")"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Name","value":"cases"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hqr"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"with"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameBuiltinPseudo","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"inl"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hq"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"=\u003e"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Keyword","value":"have"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hpq"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"q"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"NameBuiltinPseudo","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"And"},
  {"type":"NameBuiltinPseudo","value":"."},
  {"type":"Name","value":"intro"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hp"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hq"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Name","value":"apply"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Or"},
  {"type":"NameBuiltinPseudo","value":"."},
  {"type":"Name","value":"inl"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Name","value":"exact"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hpq"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameBuiltinPseudo","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"inr"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hr"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"=\u003e"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Keyword","value":"have"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hpr"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"p"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltinPseudo","value":"∧"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"r"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":":"},
  {"type":"NameBuiltinPseudo","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"And"},
  {"type":"NameBuiltinPseudo","value":"."},
  {"type":"Name","value":"intro"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hp"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hr"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Name","value":"apply"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Or"},
  {"type":"NameBuiltinPseudo","value":"."},
  {"type":"Name","value":"inr"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Name","value":"exact"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"hpr"},
  {"type":"TextWhitespace","value":"\n"}
]
